metadata-ingestion/src/datahub/cli/docker_check.py

```diff
@@ -127,16 +208,26 @@ def check_local_docker_containers(preflight_only: bool = False) -> List[str]:
 
             if container.name in ENSURE_EXIT_SUCCESS:
                 if container.status != "exited":
-                    issues.append(f"{container.name} is still running")
+                    status = ContainerStatus.STILL_RUNNING
                 elif container.attrs["State"]["ExitCode"] != 0:
-                    issues.append(f"{container.name} did not exit cleanly")
+                    status = ContainerStatus.EXIT_FAILURE
 
             elif container.status != "running":
-                issues.append(f"{container.name} is not running")
+                status = ContainerStatus.DIED
             elif "Health" in container.attrs["State"]:
                 if container.attrs["State"]["Health"]["Status"] == "starting":
-                    issues.append(f"{container.name} is still starting")
+                    status = ContainerStatus.STARTING
                 elif container.attrs["State"]["Health"]["Status"] != "healthy":
-                    issues.append(f"{container.name} is running but not healthy")
+                    status = ContainerStatus.UNHEALTHY
+
+            container_statuses.append(DockerContainerStatus(name, status))
+
+        # Check for missing containers.
+        existing_containers = {container.name for container in containers}
+        missing_containers = set(REQUIRED_CONTAINERS) - existing_containers
+        for missing in missing_containers:
+            container_statuses.append(
+                DockerContainerStatus(missing, ContainerStatus.MISSING)
+            )
 
-    return issues
+    return QuickstartStatus(container_statuses)
```

Comment:
Question - how long do we continuously poll docker in this loop before we decide to exit with the final container statuses?

Reply:
6 minutes - https://github.com/datahub-project/datahub/blob/0034b1340f90330093823fae8a44070d488ca2db/metadata-ingestion/src/datahub/cli/docker_cli.py#L742